pkg/ecdsa/tss/tss.go

```diff
@@ -169,3 +169,139 @@ func (s *ThresholdSigner) CalculateSignature(
 
 	return signature, err
 }
+
+func BroadcastRecoveryAddress(
+	parentCtx context.Context,
+	groupID string,
+	memberID MemberID,
+	groupMemberIDs []MemberID,
+	dishonestThreshold uint,
+	networkProvider net.Provider,
+	pubKeyToAddressFn func(cecdsa.PublicKey) []byte,
+	paramsBox *params.Box,
+) error {
+	const protocolReadyTimeout = 2 * time.Minute
+
+	group := &groupInfo{
+		groupID:            groupID,
+		memberID:           memberID,
+		groupMemberIDs:     groupMemberIDs,
+		dishonestThreshold: int(dishonestThreshold),
+	}
+
+	netBridge, _ := newNetworkBridge(group, networkProvider)
+	broadcastChannel, _ := netBridge.getBroadcastChannel()
+	ctx, cancel := context.WithTimeout(parentCtx, protocolReadyTimeout)
+	defer cancel()
+
+	msgInChan := make(chan *LiquidationRecoveryMessage, len(group.groupMemberIDs))
+	handleLiquidationRecoveryMessage := func(netMsg net.Message) {
+		switch msg := netMsg.Payload().(type) {
+		case *LiquidationRecoveryMessage:
+			msgInChan <- msg
+		}
+	}
+	broadcastChannel.Recv(ctx, handleLiquidationRecoveryMessage)
+
+	memberBTCAddresses := make(map[string]string)
+
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case msg := <-msgInChan:
+				for _, memberID := range group.groupMemberIDs {
+					if msg.SenderID.Equal(memberID) {
```

Comment:
Do we need to execute this check? Broadcast channel filters guarantee only signing group members can write to the channel.

Reply:
What's the difference between here and `readyProtocol` (which is where I stole the group inclusion checking from)?

Reply:
I think ready protocol was implemented before we had broadcast channel filters so we had to do this check everywhere. Need to confirm though.

Reply:
Let's make a flag to check this and not block merging.